ouroboros-consensus/test-consensus/Test/Dynamic/Network.hs

```diff
@@ -369,6 +372,45 @@ runNodeNetwork registry testBtime numCoreNodes nodeJoinPlan nodeTopology
 
       return (nodeKernel, readNodeInfo, LimitedApp app)
 
+    customProtocolCodecs
+      :: NodeConfig (BlockProtocol blk)
+      -> ProtocolCodecs blk CodecError m
+           (AnyMessage (ChainSync (Header blk) (Tip blk)))
+           Lazy.ByteString
+           Lazy.ByteString
+           (AnyMessage (TxSubmission (GenTxId blk) (GenTx blk)))
+           (AnyMessage (ChainSync blk (Tip blk)))
+           (AnyMessage (LocalTxSubmission (GenTx blk) (ApplyTxErr blk)))
+    customProtocolCodecs cfg = ProtocolCodecs
+        { pcChainSyncCodec =
+            mapFailureCodec CodecIdFailure $
+            pcChainSyncCodec protocolCodecsId
+        , pcBlockFetchCodec =
+            mapFailureCodec CodecBytesFailure $
+            pcBlockFetchCodec binaryProtocolCodecs
+        , pcBlockFetchCodecSerialised =
+            mapFailureCodec CodecBytesFailure $
+            pcBlockFetchCodecSerialised binaryProtocolCodecs
+        , pcTxSubmissionCodec =
+            mapFailureCodec CodecIdFailure $
+            pcTxSubmissionCodec protocolCodecsId
+        , pcLocalChainSyncCodec =
+            mapFailureCodec CodecIdFailure $
+            pcLocalChainSyncCodec protocolCodecsId
+        , pcLocalTxSubmissionCodec =
+            mapFailureCodec CodecIdFailure $
+            pcLocalTxSubmissionCodec protocolCodecsId
+        }
+      where
+        binaryProtocolCodecs = protocolCodecs cfg
+
+-- | Sum of 'CodecFailure' (from 'protocolCodecsId') and 'DeserialiseFailure'
+-- (from 'protocolCodecs').
+data CodecError
+  = CodecIdFailure    CodecFailure
+  | CodecBytesFailure DeserialiseFailure
+  deriving (Show, Exception)
+
```

Comment:
I solved the issues with CodecId using this custom `ProtocolCodecs` which uses the binary codec for BlockFetch but CodecId for the others.